from PyQt5 import QtCore, QtGui, QtWidgets
import sys
from PyQt5.QtWidgets import QMessageBox, QMainWindow, QStyledItemDelegate, QAction, QLineEdit


class UI_percentage(QMainWindow):
    def __init__(self, parent=None):
        super(UI_percentage, self).__init__(parent)
        # self.database_config_obj = None
    def setupUi(self):
        self.setObjectName("MainWindow")
        self.resize(520, 527)
        self.centralwidget = QtWidgets.QWidget(self)
        self.setWindowIcon(QtGui.QIcon("../icon/logo.png"))
        self.setWindowFlags(QtCore.Qt.FramelessWindowHint)
        # self.setAttribute(QtCore.Qt.WA_TranslucentBackground)
        self.centralwidget.setObjectName("centralwidget")
        self.label_3 = QtWidgets.QLabel(self.centralwidget)
        self.label_3.setGeometry(QtCore.QRect(20, 10, 501, 51))
        font = QtGui.QFont()
        font.setFamily("Segoe UI Light")
        font.setPointSize(22)
        font.setBold(False)
        font.setItalic(False)
        font.setWeight(50)
        self.label_3.setFont(font)
        self.label_3.setStyleSheet("color: rgb(11, 34, 57);")
        self.label_3.setObjectName("label_3")
        self.label = QtWidgets.QLabel(self.centralwidget)
        self.label.setGeometry(QtCore.QRect(140, 340, 421, 16))
        self.label.setObjectName("label")
        self.label_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_2.setGeometry(QtCore.QRect(30, 100, 231, 21))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.label_4 = QtWidgets.QLabel(self.centralwidget)
        self.label_4.setGeometry(QtCore.QRect(30, 200, 231, 21))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_4.setFont(font)
        self.label_4.setObjectName("label_4")
        self.label_5 = QtWidgets.QLabel(self.centralwidget)
        self.label_5.setGeometry(QtCore.QRect(30, 145, 231, 31))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_5.setFont(font)
        self.label_5.setObjectName("label_5")
        self.label_6 = QtWidgets.QLabel(self.centralwidget)
        self.label_6.setGeometry(QtCore.QRect(30, 250, 271, 21))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_6.setFont(font)
        self.label_6.setObjectName("label_6")
        self.label_7 = QtWidgets.QLabel(self.centralwidget)
        self.label_7.setGeometry(QtCore.QRect(30, 295, 281, 31))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_7.setFont(font)
        self.label_7.setObjectName("label_7")
        self.label_random = QtWidgets.QLabel(self.centralwidget)
        self.label_random.setGeometry(QtCore.QRect(350, 100, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_random.setFont(font)
        self.label_random.setObjectName("label_random")
        self.label_logic = QtWidgets.QLabel(self.centralwidget)
        self.label_logic.setGeometry(QtCore.QRect(350, 150, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_logic.setFont(font)
        self.label_logic.setObjectName("label_logic")
        self.label_liner = QtWidgets.QLabel(self.centralwidget)
        self.label_liner.setGeometry(QtCore.QRect(350, 200, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_liner.setFont(font)
        self.label_liner.setObjectName("label_liner")
        self.label_decision = QtWidgets.QLabel(self.centralwidget)
        self.label_decision.setGeometry(QtCore.QRect(350, 250, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_decision.setFont(font)
        self.label_decision.setObjectName("label_decision")
        self.label_knn = QtWidgets.QLabel(self.centralwidget)
        self.label_knn.setGeometry(QtCore.QRect(350, 300, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_knn.setFont(font)
        self.label_knn.setObjectName("label_knn")
        self.label_14 = QtWidgets.QLabel(self.centralwidget)
        self.label_14.setGeometry(QtCore.QRect(30, 390, 231, 20))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_14.setFont(font)
        self.label_14.setObjectName("label_14")
        self.label_tt = QtWidgets.QLabel(self.centralwidget)
        self.label_tt.setStyleSheet("color : red")
        self.label_tt.setGeometry(QtCore.QRect(200, 420, 291, 31))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.label_tt.setFont(font)
        self.label_tt.setObjectName("label_tt")
        self.label_random_2 = QtWidgets.QLabel(self.centralwidget)
        self.label_random_2.setGeometry(QtCore.QRect(380, 70, 131, 21))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.label_random_2.setFont(font)
        self.label_random_2.setObjectName("label_random_2")
        self.btn_close = QtWidgets.QPushButton(self.centralwidget)
        self.btn_close.setStyleSheet("background-color:rgb(218, 91, 11)")
        self.btn_close.clicked.connect(self.close)
        self.btn_close.setGeometry(QtCore.QRect(400, 470, 89, 27))
        self.btn_close.setObjectName("btn_close")
        self.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(self)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 520, 25))
        self.menubar.setObjectName("menubar")
        self.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(self)
        self.statusbar.setObjectName("statusbar")
        self.setStatusBar(self.statusbar)
        self.retranslateUi(self)
        # QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "Percentage"))
        self.label_3.setText(_translate("MainWindow", "Percentage fit for each algorithm"))
        self.label.setText(_translate("MainWindow", "*Percentage of fit is calculated on a data file of 4424 data lines."))
        self.label_2.setText(_translate("MainWindow", "1. Random Forest Classifier"))
        self.label_4.setText(_translate("MainWindow", "3. Linear Regression"))
        self.label_5.setText(_translate("MainWindow", "2. Logistic Regression"))
        self.label_6.setText(_translate("MainWindow", "4. Decision Tree Classifier"))
        self.label_7.setText(_translate("MainWindow", "5. K-Nearest Neighbors Classifier"))
        self.label_random.setText(_translate("MainWindow", "TextLabel"))
        self.label_logic.setText(_translate("MainWindow", "TextLabel"))
        self.label_liner.setText(_translate("MainWindow", "TextLabel"))
        self.label_decision.setText(_translate("MainWindow", "TextLabel"))
        self.label_knn.setText(_translate("MainWindow", "TextLabel"))
        self.label_14.setText(_translate("MainWindow", "You should choose algorithm:"))
        self.label_tt.setText(_translate("MainWindow", ""))
        self.label_random_2.setText(_translate("MainWindow", "%"))
        self.btn_close.setText(_translate("MainWindow", "Close"))


    def close(self):
        self.hide()
if __name__ == "__main__":
    app = QtWidgets.QApplication(sys.argv)
    ui = UI_percentage()
    ui.setupUi()
    ui.show()
    sys.exit(app.exec_())
